libimage/filters.go

```diff
@@ -73,25 +71,14 @@ func (r *Runtime) filterImages(ctx context.Context, images []*Image, options *Li
 // required format is `key=value` with the following supported keys:
 //
 //	after, since, before, containers, dangling, id, label, readonly, reference, intermediate
-func (r *Runtime) compileImageFilters(ctx context.Context, options *ListImagesOptions) (map[string][]filterFunc, error) {
+func (r *Runtime) compileImageFilters(ctx context.Context, options *ListImagesOptions) (filters filtersMap, needsLayerTree bool, err error) {
```

Comment:
(This works just fine as is, but I’ve been burned with named return values enough that I prefer to avoid them, or to use them _only_ for `defer` cleanup / error reporting.

Consider
```go
func baz() error { return errors.New("failure") }

func fn(situation int) (err error) {
	err = nil // why not
	switch situation {
	case 1:
		err = errors.New("failure") // OK, returns "failure"
	case 2:
		err := baz() // Easily happens, idiomatic
		if err != nil {
			break // Boom, returns "nil"
		}
		// …
	default: // nothing
	}
	return
}

func main() {
	for i := range 3 {
		fmt.Printf("situation: %v, returned %v\n", i, fn(i))
	}
}
```
)